node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/ParkedExpirer.java

```diff
@@ -0,0 +1,90 @@
+// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.hosted.provision.maintenance;
+
+import com.yahoo.config.provision.Environment;
+import com.yahoo.config.provision.NodeType;
+import com.yahoo.config.provision.Zone;
+import com.yahoo.jdisc.Metric;
+import com.yahoo.vespa.hosted.provision.Node;
+import com.yahoo.vespa.hosted.provision.NodeList;
+import com.yahoo.vespa.hosted.provision.NodeRepository;
+import com.yahoo.vespa.hosted.provision.node.Agent;
+import com.yahoo.vespa.hosted.provision.node.History;
+import com.yahoo.vespa.hosted.provision.node.Report;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.logging.Logger;
+
+import static com.yahoo.vespa.hosted.provision.maintenance.FailedExpirer.maxAllowedFailures;
+
+/**
+ * This moves expired parked nodes:
+ *
+ * - To dirty: if it does not have a hardware failure.
+ * - Otherwise the node will remain in parked.
+ *
+ * Nodes with detected hardware issues will not be recycled.
+ *
+ * @author bjormel
+ */
+public class ParkedExpirer extends NodeRepositoryMaintainer {
+
+    private static final Logger log = Logger.getLogger(ParkedExpirer.class.getName());
+
+    private final NodeRepository nodeRepository;
+    private final Duration expiry; // Grace period to allow operators to inspect
+
+    ParkedExpirer(NodeRepository nodeRepository, Zone zone, Duration interval, Metric metric) {
+        super(nodeRepository, interval, metric);
+        this.nodeRepository = nodeRepository;
+        if (zone.system().isCd()) {
+            expiry = Duration.ofMinutes(30);
+        } else {
+            if (zone.environment() == Environment.staging || zone.environment() == Environment.test) {
+                expiry = Duration.ofMinutes(10);
+            } else {
+                expiry = Duration.ofDays(4);
+            }
+        }
+    }
+
+    @Override
+    protected double maintain() {
+        NodeList allNodes = nodeRepository.nodes().list();
+        List<Node> remainingNodes = new ArrayList<>(allNodes.state(Node.State.parked)
+                .nodeType(NodeType.tenant, NodeType.host)
+                .asList());
+
+        recycleIf(node -> node.allocation().isEmpty(), remainingNodes);
+        recycleIf(node -> node.allocation().isPresent() &&
```

Comment:
You also want to make sure that neither `wantToRebuild` nor `wantToDeprovision` is set, otherwise this host may already be in the process of being deprovisioned. (And `Nodes::deallocate` with throw an exception)